src/device/v4l2/stateless/request.rs

```diff
@@ -0,0 +1,206 @@
+// Copyright 2024 The ChromiumOS Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+use std::cell::RefCell;
+use std::os::fd::AsRawFd;
+use std::rc::Rc;
+
+use v4l2r::controls::ExtControlTrait;
+use v4l2r::controls::SafeExtControl;
+use v4l2r::ioctl;
+
+use crate::device::v4l2::stateless::device::V4l2Device;
+use crate::device::v4l2::stateless::queue::V4l2CaptureBuffer;
+use crate::device::v4l2::stateless::queue::V4l2OutputBuffer;
+
+struct InitRequestHandle {
+    device: V4l2Device,
+    timestamp: u64,
+    handle: ioctl::Request,
+    buffer: V4l2OutputBuffer,
+}
+
+impl InitRequestHandle {
+    fn new(
+        device: V4l2Device,
+        timestamp: u64,
+        handle: ioctl::Request,
+        buffer: V4l2OutputBuffer,
+    ) -> Self {
+        Self {
+            device,
+            timestamp,
+            handle,
+            buffer,
+        }
+    }
+    fn ioctl<C, T>(&mut self, ctrl: C) -> &mut Self
```

Comment:
This does not tell us which ioctl we are doing. `add_ctrl` is probably a better name.